src/api/Api.js

```diff
@@ -181,11 +181,8 @@ class Api {
         // Since we are currently using PUT to save the full state back, we have to use mergeMode=REPLACE
         // to clear out existing values
         const urlForUpdate = useMergeStrategy === true ? `${url}?${getMergeStrategyParam()}` : url;
-
-        if (typeof data === 'string') {
-            return this.request('PUT', getUrl(this.baseUrl, urlForUpdate), data);
-        }
-        return this.request('PUT', getUrl(this.baseUrl, urlForUpdate), JSON.stringify(data));
+        return this.request('PUT', getUrl(this.baseUrl, urlForUpdate), String(data),
```

Comment:
Not exactly what I meant.

The PUT should be generic and still handle a JSON payload (which is probably the most frequent case).

My suggestion was to only change the line within the `if` condition and keep the rest the same:

`if (typeof data === 'string') {
    return this.request('PUT', getUrl(this.baseUrl, urlForUpdate), String(data), { headers: new Headers({ 'Content-Type': 'text/plain' }) });
    }`